### ISSUE TITLE
Black text when using with Iris and Sodium.

### ISSUE THREAD

**Issue by luotn** [maruohon/litematica#921]
When using litematica with other mods, specifically Iris and Sodium when enabling shaders, e.g. BSL Shader and Complementary Reimagined, all text in game and menus will turn black. Save and exiting from the world won't fix it too, requires full restart of Minecraft to resolve this issue.

Software Versions

> Prism Launcher: 9.1
> MC: 1.21.3
> litematica: 0.20.1
> malilib: 0.22.2
> Iris: 1.8.1+mc1.21.3
> Sodium: 0.6.3+mc1.21.3
> BSL: v8.4.01.1
> Complementary Shadres - Reimagined: r5.3

Hardware and OS:

> Apple Macbook Pro 14'inch 2021 M1 Pro 16GB ram, 8cpu + 14gpu
> macOS 15.1.1 (24B91)

![截屏2024-12-12 10 19 10](https://github.com/user-attachments/assets/3e268913-f2f6-4ec4-b428-c8cee766393f)
![截屏2024-12-12 10 19 06](https://github.com/user-attachments/assets/fb80863e-f5af-4896-a896-c97c84a7f071)
![截屏2024-12-12 10 19 03](https://github.com/user-attachments/assets/2eae4148-1cb1-4d95-9927-56dcaf8d3aa9)
 
Happy to provide more info!
PS: Thank you for your great work!

**Comment by sakura-ryoko** [comment]
This is a bug with Iris itself.  You don't need Litematica to reproduce this.

**Comment by mv-stns** [comment]
> This is a bug with Iris itself. You don't need Litematica to reproduce this.

Yes, you can minimize this issue by disabling or removing Litematica. I am also experiencing this and trying to figure out how to fix it, but I haven't found anything yet.
![scrnsht_04-01 -2025at 20 41 39](https://github.com/user-attachments/assets/ff2b01c4-1f4d-4f51-9906-c4d533bfeeb3)

**Comment by mv-stns** [comment]
Disabling the shaders in-game is not ideal, as all textures turn black. Restarting the game and disabling shaders before entering a world or server is the only way to resolve this issue.
![scrnsht_04-01 -2025at 20 45 36](https://github.com/user-attachments/assets/22751a85-4775-4a06-8aa6-12ae2300e597)

**Comment by sakura-ryoko** [comment]
Ive seen other people who aren't using Litematica report the black textures issue.  Make sure you are using the up to date Sodium / Iris combinations first.

**Comment by mv-stns** [comment]
> Ive seen other people who aren't using Litematica report the black textures issue. Make sure you are using the up to date Sodium / Iris combinations first.

Surely! Always watching for new version. I will build your latest edits in case Modrinth isn't up to date. I just found out you're managing the upper ports. Thanks a lot!

**Comment by mv-stns** [comment]
Yea upon building the latest source, the issue still persists (with litematica). When disabling litematica as said, shaders are working very good.
